internal/restorer/filerestorer.go

```diff
@@ -327,25 +333,33 @@ func (r *fileRestorer) loadBlob(rd io.Reader, blobID restic.ID, length int) ([]b
 		return nil, errors.Errorf("blob %v returned invalid hash", blobID)
 	}
 
-	return plaintext, nil
+	// move decrypted data to the start of the buffer
+	copy(buf, plaintext)
+	return buf[:len(plaintext)], nil
 }
 
-func skipReader(rd io.Reader, skip int) error {
+// skipReader reads exactly skip bytes and ignores them, reusing a buffer if given.
+func skipReader(rd io.Reader, skip int, buf []byte) ([]byte, error) {
 	const maxReadLen = 1024
```

Comment:
This is way too small. `rd.Read` can now trigger IO and is not just an in-memory copy.

Reply:
I now use `bufio` and the `Discard` method which is already defined there.